extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -181,15 +184,44 @@ export class Placement {
         }
         this.adElement_ = this.createAdElement_(baseAttributes);
         this.injector_(this.anchorElement_, this.adElement_);
-        return this.resources_.attemptChangeSize(this.adElement_,
-            TARGET_AD_HEIGHT_PX, undefined, this.margins_)
-            .then(() => {
-              this.state_ = PlacementState.PLACED;
-              return this.state_;
-            }, () => {
-              this.state_ = PlacementState.RESIZE_FAILED;
-              return this.state_;
-            });
+        if (isExperimentOn(this.ampdoc.win,
+            'amp-auto-ads-responsive')) {
+          const viewportWidth = this.resources_.getViewport().getWidth();
+          const viewportHeight = this.resources_.getViewport().getHeight();
+          const responsiveHeight =
+            getResponsiveHeightForContext_(viewportWidth, viewportHeight);
+          Services.resourcesForDoc(this.anchorElement_)
+              .getElementLayoutBox(this.anchorElement_)
+              .then(layoutBox => {
+                const direction =
+                  computedStyle(this.ampdoc.win, this.anchorElement_)
+                      ['direction'];
+                if (direction == 'rtl') {
+                  this.margins_.right = layoutBox.left;
+                } else {
+                  this.margins_.left = -layoutBox.left;
+                }
+              });
```

Comment:
As far as I can tell this code block produces a width, height and margins?
I think it would make the code more readable and easy to follow if this was broken out into a separate function that returns a typedef object with those things.

It would also mean that you don't need to duplicate the attemptChangeSize code below.
E.g.
...
this.injector_(this.anchorElement_, this.adElement_);
const adSizing = this.getSizing();
return this.resources_.attemptChangeSize(this.adElement_, adSizing.height, adSizing.width, adSizing.margins)

Reply:
Done, thanks Tom